app.ts

```diff
@@ -0,0 +1,277 @@
+      .then((data) => {
+        data = JSON.parse(data);
+        let headingData = data;
+        for (let i = 0; i < headingData.length; i++) {
+          createHeadingRow(headingData[i]);
+        }
+      });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+// Table Content (Getting the table's data)
+
+const getTable = () => {
+  try {
+    fetch("http://localhost:2050/records?from=" + paramOne + "&to=" + paramTwo, {
+      method: "GET",
+      headers: { "Content-Type": "application/json" },
+    })
+      .then((res) => res.text())
+      .then((data) => {
+        data = JSON.parse(data);
+        let contentData = data;
+        for (let i = 0; i < contentData.length; i++) {
+          createTableContent(contentData[i]);
+        }
+      });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+// Displays The Current Results Being Shown
+
+const stats = () => {
+  const pageStats: any = document.querySelector("#pageStats");
+
+  try {
+    fetch("http://localhost:2050/recordCount", {
+      method: "GET",
+      headers: { "Content-Type": "application/json" },
+    })
+      .then((res) => res.text())
+      .then((data) => {
+        data = JSON.parse(data);
+        let count = data;
+        let currentStats = "Showing results from " + paramOne + " to " + paramTwo + " out of " + count + " results.";
+        pageStats.innerHTML = currentStats;
+      });
+  } catch (error) {}
+};
+
+//// Debounce
+
+const debounce = (fn: any, delay: number) => {
+  let timer: number;
+  return function () {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      fn();
+    }, delay);
+  };
+};
+
+//// Sizing And Resizing
+
+let resizing = () => {
+  let end = paramOne + getNoOfRows();
+
+  if (end > 999999) {
+    paramTwo = 999999;
+    paramOne = paramTwo - getNoOfRows();
+  } else {
+    paramOne;
+    paramTwo = paramOne + getNoOfRows();
+  }
+  clearTable();
+  getTable();
+  stats();
+};
+
+resizing = debounce(resizing, 500);
+
+window.addEventListener("resize", resizing);
+
+//// On Window Load
+
+window.onload = function () {
+  getHeadings();
+  getTable();
+  stats();
+};
+
+//// Navigation
+
+// Next
+const nextButton: any = document.querySelector("#next");
+let nextCount = 0;
+
+const nextDebounce = (fn: any, delay: number) => {
+  let timer: any;
+  return function () {
+    nextCount++;
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      fn();
+    }, delay);
+  };
+};
+
+let next = () => {
+  if (paramTwo === 999999) {
+    alert("You have reached the final page");
+  }
+
+  let nextAmount = paramTwo - paramOne + 1;
+  let nextCountAmount = nextAmount * nextCount;
+  paramOne = paramOne + nextCountAmount;
+  paramTwo = paramOne + getNoOfRows();
+
+  let end = paramOne + getNoOfRows();
+
+  if (end > 999999) {
+    paramTwo = 999999;
+    paramOne = paramTwo - getNoOfRows();
+  }
+
+  nextCount = 0;
+
+  clearTable();
+  getTable();
+  stats();
+};
+
+next = nextDebounce(next, 500);
```

Comment:
You should not have to do this. Just pass the nextDebounce in directly to the eventListener in this case.